cluster/spanning_tree_main.cc

```diff
@@ -17,33 +17,70 @@ int getpid() { return (int)::GetCurrentProcessId(); }
 
 #include <iostream>
 #include <fstream>
-#include <string.h>
-#include <stdlib.h>
-#include <errno.h>
 
 using namespace VW;
 
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+void usage(const po::options_description& desc)
+{
+  std::cout << "usage: spanning_tree [--port,-p number] [--nondaemon] [--help,-h] [pid_file]" << std::endl;
+  std::cout << desc << std::endl;
+}
+
 int main(int argc, char* argv[])
 {
-  if (argc > 2)
+  int port = 26543;
+  bool nondaemon = false;
+
+  po::variables_map vm;
+  po::options_description desc("Spanning Tree");
+  desc.add_options()("nondaemon", po::bool_switch(&nondaemon), "Run spanning tree in foreground")(
+      "help,h", "Print help message")("port,p", po::value<int>(&port), "Port number for spanning tree to listen on");
+
+  std::string pid_file_name;
+  po::options_description hidden;
+  hidden.add_options()("pid_file", po::value<std::string>(&pid_file_name), "File to write PID value to.");
+
+  po::options_description all_options;
+  all_options.add(desc);
+  all_options.add(hidden);
+
+  po::positional_options_description pos;
+  pos.add("pid_file", -1);
+
+  try
+  {
+    po::store(po::command_line_parser(argc, argv).options(all_options).positional(pos).run(), vm);
+    po::notify(vm);
+  }
+  catch (std::exception& e)
+  {
+    std::cout << argv[0] << ": " << e.what() << std::endl << std::endl;
+    usage(desc);
+    exit(1);
```

Comment:
does this need to exit? can it `return 1`?

Reply:
Return makes sense